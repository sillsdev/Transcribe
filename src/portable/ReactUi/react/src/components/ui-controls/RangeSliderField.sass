@import "../../_variables.sass"

.RangeSliderField
    border-bottom-left-radius: 4px
    border-bottom-right-radius: 4px
    display: flex
    flex-direction: column
    height: 128px

    .SliderCaption
        display: flex
        flex-direction: row
        justify-content: center
        align-content: center
        padding: 16px
        text-align: center
        .Caption
            display: flex
            flex-direction: row
            font-family: $sans
            font-size: 16px
            line-height: 28px
            text-align: center
            letter-spacing: 0.4px
            color: $greenLight
            justify-content: center
            align-content: center

    .SliderStyle
        display: flex
        flex-direction: row
        justify-content: center
        align-content: center

    .rc-slider
        position: relative
        height: 14px
        padding: 5px 0
        width: 80%
        border-radius: 6px
        box-sizing: border-box
        -webkit-tap-highlight-color: rgba(0, 0, 0, 0)

    .rc-slider-rail
        position: absolute
        width: 100%
        background-color: $base1
        height: 2px
        border-radius: 6px

    .rc-slider-track
        position: absolute
        left: 0
        height: 2px
        border-radius: 6px

    .rc-slider-step
        position: absolute
        width: 100%
        height: 2px
        background: transparent

        .rc-slider-dot-active
            border-color: $yellowLight

        .rc-slider-dot
            position: absolute
            bottom: -3px
            margin-left: -4px
            width: 8px
            height: 8px
            background-color: $greenLight
            cursor: pointer
            border-radius: 50%
            vertical-align: middle

    .rc-slider-mark-text-active 
        color: #666

    .rc-slider-mark-text 
        position: absolute
        display: inline-block
        vertical-align: middle
        text-align: center
        cursor: pointer
        color: #999
        font-family: $sans
        font-size: 16px
        line-height: 28px
        letter-spacing: 0.4px
        top: 25px

    .rc-slider-handle
        position: absolute
        margin-left: -8px
        margin-top: -8px
        width: 16px
        height: 16px
        cursor: pointer
        cursor: -webkit-grab
        cursor: grab
        border-radius: 50%
        background-color: $yellowLight

