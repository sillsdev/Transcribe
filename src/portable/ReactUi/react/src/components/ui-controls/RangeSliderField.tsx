import Slider from 'rc-slider'
import * as React from 'react';
import './RangeSliderField.sass';

interface IProps {
    id?: string,
    caption: string,
    marks: any,
    selected: number,
    onChange?: (context: any) => any;
}

class RangeSliderField extends React.Component<IProps, any> {
    public state = {
        caption: this.props.caption,
        current: this.props.selected,
        id: this.props.id,
        marks: this.props.marks,
    }

    constructor(props: IProps) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
    }

    public handleChange(value: number) {
        this.setState({ current: value });
        if (this.props.onChange != null) {
            this.props.onChange(value)
        }
    }

    public render() {
        const { caption, marks } = this.props;
        const { current } = this.state;
        const yellowColor = "#F5CC4C";
        if(current !== null) {
            marks[current].style.color = yellowColor
        }
        return (
            <div className="RangeSliderField">
                <div className="SliderCaption"><label className="Caption">{caption}</label></div>
                <div className="SliderStyle"> <Slider min={0} max={3} marks={marks} step={null} onChange={this.handleChange} defaultValue={current} /></div>
            </div>
        )
    }

}


export default RangeSliderField;