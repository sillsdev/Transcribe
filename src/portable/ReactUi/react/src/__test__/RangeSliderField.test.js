/* jshint esversion: 6 */
import React from 'react';
import Slider from 'rc-slider'
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import RangeSliderField from '../components/ui-controls/RangeSliderField';

configure({ adapter: new Adapter() })

// Snapshot for RangeSliderField
describe('>>>Control: RangeSliderField --- Snapshot',()=>{
    const direction = "ltr";
    const yellowColor = "#F5CC4C";
    const greenColor = "#C7DE31";
    const firstIndex = direction && direction === "rtl"? {label:"synced",style:{color:greenColor},text:"Complete"}: {label:"start",style:{color:greenColor},text:"Transcribe"};
    const secondIndex = direction && direction === "rtl"? {label:"reviewed",style:{color:greenColor},text:"Upload"}: {label:"transcribed",style:{color:greenColor},text:"Review"};
    const thirdIndex = direction && direction === "rtl"? {label:"transcribed",style:{color:greenColor},text:"Review"}: {label:"reviewed",style:{color:greenColor},text:"Upload"};
    const fourthIndex = direction && direction === "rtl"? {label:"start",style:{color:greenColor},text:"Transcribe"}: {label:"synced",style:{color:greenColor},text:"Complete"};
    const minProps = {
        caption: "My Range Slider",
        marks: {0: firstIndex,1: secondIndex,2: thirdIndex,3: fourthIndex},
        selected: 1,
    };

    it('+++capturing Snapshot of RangeSliderField', () => {
        const renderedValue =  renderer.create(<RangeSliderField {...minProps}/>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('>>>Control: RangeSliderField', () => {
    let wrapper;
    const direction = "ltr";
    const yellowColor = "#F5CC4C";
    const greenColor = "#C7DE31";
    const firstIndex = direction && direction === "rtl"? {label:"synced",style:{color:greenColor},text:"Complete"}: {label:"start",style:{color:greenColor},text:"Transcribe"};
    const secondIndex = direction && direction === "rtl"? {label:"reviewed",style:{color:greenColor},text:"Upload"}: {label:"transcribed",style:{color:greenColor},text:"Review"};
    const thirdIndex = direction && direction === "rtl"? {label:"transcribed",style:{color:greenColor},text:"Review"}: {label:"reviewed",style:{color:greenColor},text:"Upload"};
    const fourthIndex = direction && direction === "rtl"? {label:"start",style:{color:greenColor},text:"Transcribe"}: {label:"synced",style:{color:greenColor},text:"Complete"};
    const minProps = {
        caption: "My Range Slider",
        marks: {0: firstIndex,1: secondIndex,2: thirdIndex,3: fourthIndex},
        selected: 1,
    };

    beforeEach(() => {
        wrapper = shallow(<RangeSliderField {...minProps} />);
    });

    it('+++ renders without exploding', () => {
        expect(wrapper.length).toEqual(1);
    });
});