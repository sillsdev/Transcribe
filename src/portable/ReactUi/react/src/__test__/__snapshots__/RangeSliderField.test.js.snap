// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control: RangeSliderField --- Snapshot +++capturing Snapshot of RangeSliderField 1`] = `
<div
  className="RangeSliderField"
>
  <div
    className="SliderCaption"
  >
    <label
      className="Caption"
    >
      My Range Slider
    </label>
  </div>
  <div
    className="SliderStyle"
  >
     
    <div
      className="rc-slider rc-slider-with-marks"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "width": "33.33333333333333%",
          }
        }
      />
      <div
        className="rc-slider-step"
      >
        <span
          className="rc-slider-dot rc-slider-dot-active"
          style={
            Object {
              "left": "0%",
            }
          }
        />
        <span
          className="rc-slider-dot rc-slider-dot-active"
          style={
            Object {
              "left": "33.33333333333333%",
            }
          }
        />
        <span
          className="rc-slider-dot"
          style={
            Object {
              "left": "66.66666666666666%",
            }
          }
        />
        <span
          className="rc-slider-dot"
          style={
            Object {
              "left": "100%",
            }
          }
        />
      </div>
      <div
        aria-disabled={false}
        aria-valuemax={3}
        aria-valuemin={0}
        aria-valuenow={1}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "33.33333333333333%",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      >
        <span
          className="rc-slider-mark-text rc-slider-mark-text-active"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "color": "#C7DE31",
              "left": "0%",
              "marginLeft": "-15.000000000000002%",
              "width": "30.000000000000004%",
            }
          }
        >
          start
        </span>
        <span
          className="rc-slider-mark-text rc-slider-mark-text-active"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "color": "#F5CC4C",
              "left": "33.33333333333333%",
              "marginLeft": "-15.000000000000002%",
              "width": "30.000000000000004%",
            }
          }
        >
          transcribed
        </span>
        <span
          className="rc-slider-mark-text"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "color": "#C7DE31",
              "left": "66.66666666666666%",
              "marginLeft": "-15.000000000000002%",
              "width": "30.000000000000004%",
            }
          }
        >
          reviewed
        </span>
        <span
          className="rc-slider-mark-text"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "color": "#C7DE31",
              "left": "100%",
              "marginLeft": "-15.000000000000002%",
              "width": "30.000000000000004%",
            }
          }
        >
          synced
        </span>
      </div>
    </div>
  </div>
</div>
`;
